nixos/modules/security/auditd.nix

```diff
@@ -1,18 +1,27 @@
-{ config, lib, pkgs, ... }:
-
-with lib;
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
 
 {
-  options.security.auditd.enable = mkEnableOption "the Linux Audit daemon";
+  options.security.auditd.enable = lib.mkEnableOption "the Linux Audit daemon";
 
-  config = mkIf config.security.auditd.enable {
+  config = lib.mkIf config.security.auditd.enable {
     boot.kernelParams = [ "audit=1" ];
 
     environment.systemPackages = [ pkgs.audit ];
 
     systemd.services.auditd = {
       description = "Linux Audit daemon";
       wantedBy = [ "basic.target" ];
+      wants = [
```

Comment:
We shouldn't need wants here. Setting only `after` should be enough.

Reply:
Even with `DefaultDependencies = no` ? How do we make sure then that this service is not started before the tmpfiles one got added to the transaction? Or does the whole transaction always get created before anything starts executing?

Reply:
tmpfiles is pulled in by sysinit.target. By the time basic.target is started, tmpfiles should've already been pulled in. Then it should only be an ordering issue and thus only require `after`. Also compare the upstream service config: https://github.com/linux-audit/audit-userspace/blob/master/init.d/auditd.service.in#L19

In general it seems that auditd should probably be:

```nix
wantedBy = [ "sysinit.target" ];
before = [ "sysinit.target" ];
after = [ "systemd-tmpfiles-setup.service" ];
```

Reply:
Done. There's a couple of other things that we might want to take from the upstream unit then, but I'll do that as a separate PR.